### PR TITLE
allow using config files on AWS providers (ports #875)

### LINKED ISSUE #909: Deploying to S3 should read from ~/.aws/credentials file if access_key_id and secret_access_key are not provided
Hi when I try to deploy with s3 my team does not have the access and secret key. However it is on instance's ~/.aws/credentials file that travis is hosted on. Can we add the feature to read from that file before reading the access_key_id and secret_access_key in the .yml file.

here is the error message I get:
```
Preparing deploy
missing access_key_id
failed to deploy
```

### PR DESCRIPTION
**Pull request by svenfuchs**
This resolves #909

### CODE REVIEW

**Comment by BanzaiMan** [comment]
As with #875, this should be tested.

File: spec/dpl/providers/codedeploy_spec.rb [hunk 1]
```diff
@@ -41,60 +41,91 @@
...
   end
 
-  describe 'given --wait_until_deployed' do
-    it { should create_deployment revision: github_revision }
-    it { should have_run '[print] Waiting for the deployment to finish ' }
-    it { should get_deployment }
+  describe 'with ~/.aws/config' do
+    let(:args) { |e| %w(--access_key_id id --secret_access_key secret) }
+
+    file '~/.aws/config', <<~str
```
Comment (BanzaiMan) [inline spec/dpl/providers/codedeploy_spec.rb]: `<<~` is not valid in Ruby 2.2.x, so we need a different here-doc marker.

Comment (svenfuchs) [reply spec/dpl/providers/codedeploy_spec.rb]: true. i keep forgetting that 😞

Comment (svenfuchs) [reply spec/dpl/providers/codedeploy_spec.rb]: done!

**Pull request merged**